src/Integration.Vsix/SonarLintTagger/TextBufferIssueTracker.cs

```diff
@@ -121,38 +125,56 @@ public void RemoveTagger(IssueTagger tagger)
             {
                 // Last tagger was disposed of. This is means there are no longer any open views on the buffer so we can safely shut down
                 // issue tracking for that buffer.
-                document.FileActionOccurred -= FileActionOccurred;
-                textBuffer.ChangedLowPriority -= OnBufferChange;
+                document.FileActionOccurred -= SafeOnFileActionOccurred;
+                textBuffer.ChangedLowPriority -= SafeOnBufferChange;
                 textBuffer.Properties.RemoveProperty(typeof(TextBufferIssueTracker));
                 Provider.RemoveIssueTracker(this);
             }
         }
 
-        private void FileActionOccurred(object sender, TextDocumentFileActionEventArgs e)
+        private void SafeOnFileActionOccurred(object sender, TextDocumentFileActionEventArgs e)
         {
-            if (e.FileActionType == FileActionTypes.DocumentRenamed)
+            // Handles callback from VS. Suppress non-critical errors to prevent them
+            // propagating to VS, which would display a dialogue and disable the extension.
```

Comment:
We didn't manage to repro the error @loic-joly-sonarsource reported in #777.
However, we can log more info about any errors and prevent them from bubbling up to VS, which otherwise would display a dialogue and disable the extension.

Suppress exception shouldn't cause any major issues here; worst case is that the issue tagging for the current file will disappear/be wrong until it is refreshed by the file being saved.

Reply:
I've added try-catch blocks round the only two callbacks from VS in this class.